common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -206,6 +208,17 @@ def handle_ajax(self, dispatch, data):  # TODO: bounds checking
                 self.position = 1
             return json.dumps({'success': True})
 
+        if dispatch == 'get_completion':
+            usage_key = data.get('usage_key', None)
+            item = self.get_child(UsageKey.from_string(usage_key))
+            if item:
+                completion_service = self.runtime.service(self, 'completion')
+                complete = completion_service.get_completion_for_vertical(item) if self.is_user_authenticated else None
```

Comment:
#NIT Seems like you can check `self.is_user_authenticated` condition even earlier than this. Maybe check it before the `self.get_child()` call?